### PR TITLE
Print the seed of CGAL::default_random

### PR DESCRIPTION
**Pull request by afabri**
when CGAL_TEST_SUITE   or CGAL_PRINT_SEED is defined

Rationale: Again and again we get timeouts in the [testsuite](https://cgal.geometryfactory.com/CGAL/Members/testsuite/CGAL-4.8-Ic-145/Point_set_shape_detection_3_Examples/TestReport_lrineau_CentOS5.gz),  and cannot reproduce them as we do not know the seed of `CGAL::default_random`.    

With this PR we dump the seed in the testsuite, and can advice users with a bug report to add a define.

### CODE REVIEW

File: Random_numbers/include/CGAL/Random.h [hunk 1]
```diff
@@ -237,7 +238,8 @@ CGAL_EXPORT extern  Random  default_random;
 #ifdef CGAL_HEADER_ONLY
 inline Random& get_default_random()
 {
-  static Random default_random;
+  static Random default_random("default");
+  std::cerr << "get_default_random()" << std::endl;
```
Comment (sloriot) [inline Random_numbers/include/CGAL/Random.h]: This output is not protected by any macro

**Comment by afabri** [comment]
@lrineau  Would that be ok for 4.8?    It is rather trivial and potentially makes it easier to reproduce bugs.

**Comment by lrineau** [comment]
As the testsuite only tests with `CGAL_TEST_SUITE` defined, tests should be done outside the testsuite. If we want to avoid retesting fully CGAL-4.8 (already tested in CGAL-4.8-beta1), maybe that PR should be delayed until the branch for CGAL-4.8 is created.

**Comment by afabri** [comment]
We could for one testsuite remove the   `defined(CGAL_TEST_SUITE)`

**Comment by lrineau** [comment]
I have made adjustements, with the commit https://github.com/CGAL/cgal/pull/843/commits/79f54a6c7c3af0a1aa1a6e845990fd06f950f63b, because that PR will change the ABI of `libCGAL`.

I have also added two trivial tests.

**Comment by lrineau** [comment]
Will be tested in [CGAL-4.9-Ic-5][5].

[5]: https://cgal.geometryfactory.com/CGAL/testsuite/results-4.9-Ic-5.shtml

**Comment by lrineau** [comment]
There is a design bug. A compilation of all `integration` on my machine ends up with that compilation error:
```
/home/lrineau/Git/cgal/Mesh_3/include/CGAL/Labeled_mesh_domain_3.h: In constructor ‘CGAL::Labeled_mesh_domain_3<Function, BGT>::Labeled_mesh_domain_3(const F&, const Sphere_3&, const FT&, CGAL::Random*)’:
/home/lrineau/Git/cgal/Mesh_3/include/CGAL/Labeled_mesh_domain_3.h:520:32: error: call of overloaded ‘Random(int)’ is ambiguous
     p_rng_ = new CGAL::Random(0);
                                ^
In file included from /home/lrineau/Git/cgal/Generator/include/CGAL/generators.h:34:0,
                 from /home/lrineau/Git/cgal/Generator/include/CGAL/point_generators_3.h:29,
                 from /home/lrineau/Git/cgal/Triangulation_3/include/CGAL/Regular_triangulation_3.h:32,
                 from /home/lrineau/Git/cgal/Mesh_3/include/CGAL/Mesh_triangulation_3.h:30,
                 from /home/lrineau/Git/cgal/Mesh_3/examples/Mesh_3/mesh_implicit_domains_2.cpp:3:
/home/lrineau/Git/cgal/Random_numbers/include/CGAL/Random.h:69:44: note: candidate: CGAL::Random::Random(char*)
/home/lrineau/Git/cgal/Random_numbers/include/CGAL/Random.h:68:44: note: candidate: CGAL::Random::Random(unsigned int)
/home/lrineau/Git/cgal/Random_numbers/include/CGAL/Random.h:48:7: note: candidate: CGAL::Random::Random(const CGAL::Random&)
 class Random {
       ^
```

**Comment by lrineau** [comment]
Will be tested in [CGAL-4.9-Ic-5][5].

[5]: https://cgal.geometryfactory.com/CGAL/testsuite/results-4.9-Ic-5.shtml

File: Random_numbers/include/CGAL/Random.h [hunk 2]
```diff
@@ -237,7 +237,7 @@ CGAL_EXPORT extern  Random  default_random;
 #ifdef CGAL_HEADER_ONLY
 inline Random& get_default_random()
 {
-  static Random default_random;
+  static Random default_random("default");
```
Comment (lrineau) [inline Random_numbers/include/CGAL/Random.h]: @afabri This modification should be canceled too.

**Comment by lrineau** [comment]
It seems to work pretty well. For example:
https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-4.9-Ic-5/Triangulation_2/TestReport_lrineau_Debian-Stable.gz

TODO: I should remove my bump of the SONAME minor version.

**Comment by lrineau** [comment]
TODO done. See  fd4e0a5.

I have also fixed a bug in d2b0900.

**Comment by lrineau** [comment]
Will be re-rested tonight.

File: Random_numbers/include/CGAL/Random_impl.h [hunk 3]
```diff
@@ -46,7 +46,11 @@ Random( )
     std::time_t s;
     std::time( &s);
     seed = (unsigned int)s;
-
+#if defined( CGAL_TEST_SUITE ) || defined( CGAL_PRINT_SEED )
+    if(this == & get_default_random()){
+      std::cerr << "default_random::get_seed() = " << seed << std::endl;
```
Comment (cjamin) [inline Random_numbers/include/CGAL/Random_impl.h]: Shouldn't it be: `"get_default_random()::get_seed() = "`... ?

Comment (lrineau) [inline Random_numbers/include/CGAL/Random_impl.h]: Yes, I agree.

Comment (lrineau) [inline Random_numbers/include/CGAL/Random_impl.h]: Fixed by #975

**Pull request merged**